rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnUnionTypeRector/Fixture/extended_class.php.inc

```diff
@@ -26,7 +26,7 @@ use Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnUnionTypeRector\Source
 
 final class ExtendedClass extends SomeParentClass
 {
-    public function run(): \Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnUnionTypeRector\Source\SomeParentClass
+    public function run(): \Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnUnionTypeRector\Fixture\ExtendedClass|\Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnUnionTypeRector\Source\SomeParentClass
```

Comment:
this was a feature, it should be wiped into single `SomeParentClass` as before, as other type is child of `SomeParentClass` only.

if we have removing feature here, the new rule to make it explicit should be exists

https://3v4l.org/LW9LU

Reply:
This was a good way to use return types early back before union types. It was incorrect as it included all other child classes of `SomeParentClass`. Basically we broke type precission for the user :)

Now since PHP 8.0+ is widely used, union type is preffered as more exact.